lib/datadog/core/configuration/option.rb

```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
```

Comment:
Meta note on this PR: This is the kind of PR that has the potential to have undetected conflicts with other ongoing PRs -- e.g. any PRs that touch settings that branched out before this PR have the potential to cause issues.

Here's my suggestion -- before merging this PR to master:

1. Diff master and the base of the `core-configuration-add-env-var-and-casting-functionality` branch, and see if there's any changes that haven't been picked up
2. (if needed) Merge master into this branch, include any updates
3. Do a quick round of open PRs and see if any touch the settings; If they do, ask them to hold off until this gets merged, and after that, ask them to rebase on top of master.

Reply:
> Meta note on this PR

Meta-question: this has nothing to do with the `# frozen_string_literal: true` comment, and only uses that first line as anchor for a comment, correct?

Meta-suggestion: maybe it should go as a review comment in "Finish your review"? but then it's not a "threaded" discussion I guess (subsequent discussion would happen in the global PR discussion tab).

Reply:
> Meta-suggestion: maybe it should go as a review comment in "Finish your review"? but then it's not a "threaded" discussion I guess (subsequent discussion would happen in the global PR discussion tab).

Yes, it's for the whole review, and I only picked elsewhere because I find people sometimes miss the comment posted with the review itself, and I thought this comment was important enough not to be missed.

Plus, as you pointed out, it's really annoying to not have threading for those comments.

So I'm just working around GitHub UX limitations >_>